cpp/dolfinx/fem/Expression.h

```diff
@@ -112,11 +114,43 @@ class Expression
     // Create data structures used in evaluation
     std::vector<double> coordinate_dofs(3 * num_dofs_g);
 
+    int num_argument_dofs = 1;
+    xtl::span<const std::uint32_t> cell_info;
+    std::function<void(const xtl::span<T>&,
+                       const xtl::span<const std::uint32_t>&, std::int32_t,
+                       int)>
+        dof_transform_to_transpose
+        = [](const xtl::span<T>&, const xtl::span<const std::uint32_t>&,
+             std::int32_t, int)
+    {
+      // Do nothing
+    };
+
+    if (_argument_function_space)
+    {
+      num_argument_dofs
+          = _argument_function_space->dofmap()->element_dof_layout().num_dofs();
+      const auto element = _argument_function_space->element();
+
+      if (element->needs_dof_transformations())
+      {
+        _mesh->topology_mutable().create_entity_permutations();
+        cell_info = xtl::span(_mesh->topology().get_cell_permutation_info());
+        dof_transform_to_transpose
+            = element
+                  ->template get_dof_transformation_to_transpose_function<T>();
```

Comment:
What's does `template` do here?

Reply:
ISO C++03 14.2/4:

When the name of a member template specialization appears after . or -> in a postfix-expression, or after nested-name-specifier in a qualified-id, and the postfix-expression or qualified-id explicitly depends on a template-parameter (14.6.2), the member template name must be prefixed by the keyword template. Otherwise the name is assumed to name a non-template.